docs/guides/beginner-typescript.md

```diff
@@ -0,0 +1,484 @@
+---
+title: Beginner TypeScript Guide
+nav: 7
+---
+
+Zustand is a lightweight state manager, particularly used with React. Zustand avoids reducers, context, and boilerplate.
+Paired with TypeScript, you get a strongly typed store-state, actions, and selectors-with autocomplete and compile-time safety.
+
+In this basic guide we’ll cover:
+
+- Creating a typed store (state + actions)
+- Using the store in React components with type safety
+- Structuring and coordinating multiple stores
+- Resetting the store safely with types
+- Extracting and reusing Store type (for props, tests, and utilities)
+- Composing multiple selectors and building derived state (with type inference and without extra re-renders)
+- Middlewares with TypeScript support (`combine`, `devtools`, `persist`)
+- Async actions with typed API responses
+- Using `useStore`, `useStoreWithEqualityFn` and `useShallow` hooks with Typescript
+- Additional API (`createStore`, `createWithEqualityFn`)
+
+### Creating a Store with State & Actions
+
+Here we describe state and actions using an Typescript interface. The `<BearState>` generic forces the store to match this shape.
+This means if you forget a field or use the wrong type, TypeScript will complain. Unlike plain JS, this guarantees type-safe state management.
+The `create` function uses the curried form, which results in a store of type `UseBoundStore<StoreApi<BearState>>`.
+
+```ts
+// store.ts
+import { create } from 'zustand'
+
+// Define types for state & actions
+interface BearState {
+  bears: number
+  food: string
+  feed: (food: string) => void
+}
+
+// Create store using the curried form of `create`
+export const useBearStore = create<BearState>()((set) => ({
+  bears: 2,
+  food: 'honey',
+  feed: (food) => set(() => ({ food })),
+}))
+```
+
+### Using the Store in Components
+
+Inside components, you can read state and call actions. Selectors `(s) => s.bears` subscribe to only what you need.
+This reduces re-renders and improves performance. JS can do this too, but with TS your IDE autocompletes state fields.
+
+```ts
+import { useBearStore } from './store'
+
+function BearCounter() {
+  // Select only "bears" to avoid unnecessary re-renders
+  const bears = useBearStore((s) => s.bears)
+  return <h1>{bears} bears around</h1>
+}
+
+function BearFeeder() {
+  // Destructure multiple fields if needed
+  const { food, feed } = useBearStore()
+  return (
+    <button onClick={() => feed("berries")}>
+      Feed bears {food}
+    </button>
+  )
+}
```

Comment:
Do we need to show this usage in this guide? Usually, people get confused with this usage because it will rerender with other (hidden) value changing.